cli/config/credentials/file_store.go

```diff
@@ -52,7 +52,11 @@ func (c *fileStore) GetAll() (map[string]types.AuthConfig, error) {
 
 // Store saves the given credentials in the file store.
 func (c *fileStore) Store(authConfig types.AuthConfig) error {
-	c.file.GetAuthConfigs()[authConfig.ServerAddress] = authConfig
+	authConfigs := c.file.GetAuthConfigs()
+	if authConfigs == nil {
+		authConfigs = make(map[string]types.AuthConfig)
+	}
```

Comment:
Instead of doing the nil check on caller side, I think it would be better to solve this issue in the `GetAuthConfigs` directly:

https://github.com/docker/cli/blob/df04aca5d22171a20d042a1ed289f0ae1c2acb48/cli/config/configfile/file.go#L95-L98


This way we won't have to do the nil check in all the places GetAuthConfigs is called (there are currently no other places which break with nil value though).